source/engine/graphics/NcGraphicsImpl.cpp

```diff
@@ -37,28 +37,43 @@ namespace
         nc::graphics::DebugRenderer debugRenderer;
         #endif
     };
+
+    auto ToVulkanApi(nc::graphics::GraphicsApi api) -> uint32_t
+    {
+        switch(api)
+        {
+            case nc::graphics::GraphicsApi::Vulkan_1_0: return VK_API_VERSION_1_0;
+            case nc::graphics::GraphicsApi::Vulkan_1_1: return VK_API_VERSION_1_1;
+            case nc::graphics::GraphicsApi::Vulkan_1_2: return VK_API_VERSION_1_2;
+        }
+
+        throw nc::NcError{"Unknown GraphicsApi"};
+    }
 } // anonymous namespace
 
 namespace nc::graphics
 {
-    auto BuildGraphicsModule(bool enableModule, Registry* reg, const nc::GpuAccessorSignals& gpuAccessorSignals, window::WindowImpl* window) -> std::unique_ptr<NcGraphics>
+    auto BuildGraphicsModule(bool enableModule, Registry* registry, GraphicsInitInfo info, window::WindowImpl* window) -> std::unique_ptr<NcGraphics>
     {
         if (enableModule)
         {
             NC_LOG_TRACE("Creating NcGraphics module");
-            return std::make_unique<NcGraphicsImpl>(reg, gpuAccessorSignals, window);
+            return std::make_unique<NcGraphicsImpl>(registry, std::move(info), window);
         }
 
         NC_LOG_TRACE("Creating NcGraphics module stub");
-        return std::make_unique<NcGraphicsStub>(reg);
+        return std::make_unique<NcGraphicsStub>(registry);
     }
 
-    NcGraphicsImpl::NcGraphicsImpl(Registry* registry, const nc::GpuAccessorSignals& gpuAccessorSignals, window::WindowImpl* window)
+    NcGraphicsImpl::NcGraphicsImpl(Registry* registry, GraphicsInitInfo info, window::WindowImpl* window)
         : m_registry{ registry },
-          m_camera{},
-          m_graphics{ &m_camera,
-                      registry,
-                      gpuAccessorSignals,
+          // TODO #341: Instead of constructing here, pass in from BuildGraphicsModule
```

Comment:
I like referencing the issue #, but would prefer we link it (either by URL or otherwise)